src/Model.elm

```diff
@@ -498,3 +504,88 @@ update msg model =
+                                -}
+                                oldNotifications =
+                                    model.notifications
+
+                                {-
+                                   @FIXME: we should add a function to `Mastodon`
+                                   with this typeSignature :
+                                   Notification -> List NotificationAggregate -> List NotificationAggregate
+                                -}
+                                notificationAggregate =
+                                    Mastodon.NotificationAggregate
+                                        notification.type_
+                                        notification.status
+                                        [ notification.account ]
+                                        notification.created_at
+                            in
+                                { model | notifications = notificationAggregate :: oldNotifications } ! []
+
+                        Err error ->
+                            { model | errors = error :: model.errors } ! []
+
+                Mastodon.StatusResult result ->
+                    case result of
+                        Ok status ->
+                            let
+                                {-
+                                   Limitation of Elm where you can't reference
+                                   model.notifications inside a { model | …)
+                                -}
+                                oldLocalTimeline =
+                                    model.userTimeline
+                            in
+                                { model | userTimeline = status :: oldLocalTimeline } ! []
+
+                        Err error ->
+                            { model | errors = error :: model.errors } ! []
+
+        NewWebsocketLocalMessage message ->
+            -- @TODO
+            model ! []
+
+        NewWebsocketGlobalMessage message ->
+            -- @TODO
+            model ! []
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.batch <|
+        case model.client of
+            Just client ->
+                [ Mastodon.subscribeToWebSockets
+                    client
+                    Mastodon.UserStream
+                    NewWebsocketUserMessage
+                ]
+                    ++ (if model.useGlobalTimeline then
+                            [ Mastodon.subscribeToWebSockets
+                                client
+                                Mastodon.GlobalPublicStream
+                                NewWebsocketGlobalMessage
+                            ]
+                        else
+                            [ Mastodon.subscribeToWebSockets
+                                client
+                                Mastodon.LocalPublicStream
+                                NewWebsocketLocalMessage
+                            ]
+                       )
```

Comment:
Initially I did subscribe to all Websockets, but for the global timeline, it's a lot of stuff to handle for something the user may never look at.
I think we should subscribe to updates via WebSockets only if the timeline is displayed. And on first display, we just always initialize the timeline using a classic REST Call.